js/ui/map.js

```diff
@@ -70,7 +70,11 @@ var defaultMaxZoom = 20;
  */
 var Map = module.exports = function(options) {
 
-    options = this.options = util.inherit(this.options, options);
+    options = util.inherit(this.options, options);
+    this._container = options.container;
```

Comment:
This should handle the id case so that the meaning `this._container` doesn't change depending on whether `_setupContainer` was called or not.